pkg/storage/azureblob/azureblob.go

```diff
@@ -53,18 +55,60 @@ func New(conf *config.AzureConfig) (*Storage, error) {
 	return &Storage{cl: cl, conf: conf}, nil
 }
 
-// Save implements the (github.com/gomods/athens/pkg/storage).Saver interface.
-func (s *Storage) Save(ctx context.Context, module, version string, mod []byte, zip io.Reader, info []byte) error {
-	const op errors.Op = "azure.Save"
-	ctx, span := observ.StartSpan(ctx, op.String())
-	defer span.End()
-	err := moduploader.Upload(ctx, module, version, bytes.NewReader(info), bytes.NewReader(mod), zip, s.cl.UploadWithContext, s.conf.TimeoutDuration())
+// BlobExists checks if a particular blob exists in the container
+func (c *azureBlobStoreClient) BlobExists(ctx context.Context, path string) (bool, error) {
+	// TODO: Any better way of doing this ?
+	blobURL := c.containerURL.NewBlockBlobURL(path)
+	_, err := blobURL.GetProperties(ctx, azblob.BlobAccessConditions{})
+	if err != nil {
+		serr := err.(azblob.StorageError)
+		if serr.Response().StatusCode == 404 {
```

Comment:
Can you do something like `if serr, ok := err.(azblob.StorageError); !ok {...` and also use `http.StatusNotFound` instead of 404?